src/nouveau/src/nouveau_gun.erl

```diff
@@ -36,20 +36,32 @@
 
 -define(NOUVEAU_HOST_HEADER, nouveau_host_header).
 
+-record(state, {
+    enabled,
+    url
+}).
+
 start_link() ->
     gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
 
 host_header() ->
     persistent_term:get(?NOUVEAU_HOST_HEADER).
 
 init(_) ->
+    Enabled = nouveau:enabled(),
     URL = nouveau_util:nouveau_url(),
-    case start_gun(URL) of
-        {ok, _PoolPid} ->
-            ok = config:listen_for_changes(?MODULE, URL),
-            {ok, nil};
-        {error, Reason} ->
-            {error, Reason}
+    State = #state{enabled = Enabled, url = URL},
+    ok = config:listen_for_changes(?MODULE, State),
+    if
+        Enabled ->
+            case start_gun(URL) of
+                {ok, _PoolPid} ->
+                    {ok, nil};
+                {error, Reason} ->
+                    {error, Reason}
+            end;
+        true ->
+            {ok, nil}
```

Comment:
Since we have to run basically the same logic in init and on reconfigure maybe just have a single maybe_start() or reconfigure() function that keeps track of the started `Url` and on reconfigure we check:
```erlang
case {Enabled, StartedUrl} of
   {true, undefined} -> start(), set StartedUrl = Url;
   {true, StartedUrl != Url, Url!=undefined} -> stop(), start(), set StartedUrl = Url;
   {false, StartedUrl != undefined} -> stop(), set StartedUrl = undefined;
   {true, Url = StartedUrl} -> ok;
   {false, _} ok
}
 ```